src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,158 @@
+            }
+        }
+        return output.toString();
+    }
+
+    /**
+     * Takes in user inputs.
+     *
+     * @return Input by the user.
+     */
+    public String readCommand() {
+        String input = sc.nextLine().trim();
+        return input;
+    }
+
+    /**
+     * Displays text when the program stops.
+     *
+     * @return String containing message to terminate.
+     */
+    public String showBye() {
+        return " I will be back! ";
+    }
+
+    /**
+     * Displays error message.
+     *
+     * @param message Error message.
+     * @return String containing error message.
+     */
+    public String showError(String message) {
+        return message;
+    }
+
+    /**
+     * Displays divider.
+     *
+     * @return String containing a divider.
+     */
+    public String showLine() {
+        return "\n ******************************************************************** \n";
+    }
+
+    /**
+     * Displays the list of matching task in the TaskList.
+     *
+     * @param tasks ArrayList of task.
+     * @return String that contains list of matching tasks.
+     */
+    public String showMatchingTask(ArrayList<Task> tasks) {
+        StringBuilder output = new StringBuilder(" Here are the matching tasks in your list:\n");
+        for (int i = 0; i < tasks.size(); i++) {
+            int count = i + 1;
+            output.append(String.format("   %d. ", count)).append(tasks.get(i).toString());
+        }
+        return output.toString();
+    }
+
+    /**
+     * Displays the task that is completed.
+     *
+     * @param task Task that is completed.
+     * @return String that contains the task that is done.
+     */
+    public String showDoneTask(Task task) {
+        return " Nice! Target Eliminated: \n   "
+                + task.toString();
+    }
+
+    /**
+     * Displays the task that has been undone.
+     *
+     * @param task Task that is undone.
+     * @return String that contains the task that has been undone.
+     */
+    public String showUndoTask(Task task) {
+        return " Task has been undone: \n   "
+                + task.toString();
+    }
+
+    /**
+     * Displays the task that is deleted.
+     *
+     * @param task Task that is deleted.
+     * @param numTask Number of tasks.
+     * @return String that contains the deleted task.
+     */
+    public String showDeletedTask(Task task, int numTask) {
+        return String.format(" Noted. Target Scraped: \n   %s \n "
+                + "Now you have %d tasks in the list. ", task.toString(), numTask);
+    }
+
+    /**
+     * Displays the task that is added.
+     *
+     * @param task Task that is added.
+     * @param numTask Number of tasks.
+     * @return String containing the task added.
+     */
+    public String showTask(Task task, int numTask) {
+        String output = new StringBuilder().append(" Got it. I've added this task: \n")
```

Comment:
Maybe you could separate this (eg. by assigning String.format to a variable first), so the line wouldn't be too long.

Reply:
oh okay sure!